src/coreclr/src/gc/env/gcenv.object.h

```diff
@@ -164,7 +164,11 @@ class Object
 
     MethodTable * GetGCSafeMethodTable() const
     {
+#ifdef HOST_64BIT
+        return (MethodTable *)((uintptr_t)m_pMethTab & ~7);
```

Comment:
This will also need update in https://github.com/dotnet/diagnostics/blob/2e2e6775d46a31e44218d62657b4169fa13ac74d/src/SOS/Strike/sos.cpp#L122 and potentially other places that have this mask hardcoded.

Reply:
thanks @jkotas! we will be sure to change this (in another PR). for now I wanted to check this in so Peter can continue working on it while I take care of some other urgent workitem.